openvdb_houdini/ParmFactory.cc

```diff
@@ -1047,9 +1069,200 @@ struct OpFactory::Impl
 
+        mergeSpareData(op->spareData(), mSpareData);
+
         return op;
     }
 
+    void initScripting()
+    {
+        // Install an HScript command to retrieve spare data from operators.
+        if (auto* cmgr = CMD_Manager::getContext()) {
+            if (!cmgr->isCommandDefined(kSpareDataCmdName)) {
+                cmgr->installCommand(kSpareDataCmdName, "", cmdGetOperatorSpareData);
+            }
+        }
+
+        // Install Python functions to retrieve spare data from operators.
+        static bool sDidInstallHOMModule = false;
+        if (!sDidInstallHOMModule) {
+            // Install a _dwhoudiniutils module with a NodeType_spareData() function.
+            static PY_PyMethodDef sMethods[] = {
+                {"NodeType_spareData", homGetOperatorSpareData, PY_METH_VARARGS(), ""},
+                { nullptr, nullptr, 0, nullptr }
+            };
+            {
+                PY_InterpreterAutoLock interpreterLock;
+                PY_Py_InitModule("_dwhoudiniutils", sMethods);
+                sDidInstallHOMModule = true;
+            }
+
+            // Add methods to the hou.NodeType class.
+            PYrunPythonStatementsAndExpectNoErrors("\
+def _spareData(self, name):\n\
+    '''\n\
+    spareData(name) -> str or None\n\
+    \n\
+    Return the spare data with the given name, or None\n\
+    if no data with that name is defined for this node type.\n\
+    \n\
+    Currently, only node types defined with OpenVDB's OpFactory\n\
+    can have spare data.  See www.openvdb.org for more information.\n\
+    '''\n\
+    import _dwhoudiniutils\n\
+    return _dwhoudiniutils.NodeType_spareData(self.category().name(), self.name(), name)\n\
+\n\
+def _spareDataDict(self):\n\
+    '''\n\
+    spareDataDict() -> dict of str to str\n\
+    \n\
+    Return a dictionary of the spare data for this node type.\n\
+    \n\
+    Currently, only node types defined with OpenVDB's OpFactory\n\
+    can have spare data.  See www.openvdb.org for more information.\n\
+    '''\n\
+    import _dwhoudiniutils\n\
+    return _dwhoudiniutils.NodeType_spareData(self.category().name(), self.name())\n\
+\n\
+nt = __import__('hou').NodeType\n\
+nt.spareData = _spareData\n\
+nt.spareDataDict = _spareDataDict\n\
+del nt, _spareData, _spareDataDict\n");
+        }
+    }
+
+    // HScript callback to retrieve spare data from an OP_OperatorDW-derived operator
+    static void cmdGetOperatorSpareData(CMD_Args& args)
+    {
+        // The operator's network type ("Sop", "Dop", etc.)
+        const char* const networkType = args[1];
+        // The operator's name
+        const char* const opName = args[2];
+        // An optional spare data token
+        const char* const token = args[3];
+
+        if (!networkType || !opName) {
+            /// @todo Install this as a command.help file?
+            args.out() << kSpareDataCmdName << "\n\
+\n\
+    List spare data associated with an operator type.\n\
```

Comment:
I know this isn't the right way to document an HScript command, but I really don't want to move these twenty lines to a separate `command.help` file.